functions/index.js

```diff
@@ -1,8 +1,40 @@
 const functions = require('firebase-functions');
+const admin = require('firebase-admin');
 
-// // Create and Deploy Your First Cloud Functions
-// // https://firebase.google.com/docs/functions/write-firebase-functions
-//
-exports.helloWorld = functions.https.onRequest((request, response) => {
-  response.send('Hello from Firebase!');
-});
+admin.initializeApp();
+const db = admin.firestore();
+const fieldValue = admin.firestore.FieldValue;
+
+exports.onFollow = functions.firestore
+  .document('/users/{userId}/followers/{followerId}')
+  .onCreate((snapshot, context) => {
+    const { userId } = context.params;
+    const currentUser = snapshot.data().ref;
+    return db.collection('users').doc(currentUser.id).collection('followed')
+      .add({
+        ref: db.collection('users').doc(userId),
+        created_at: fieldValue.serverTimestamp(),
+      });
+  });
+
+const getFolloweddByUserId = (id, currentUser) => db.collection('users')
+  .doc(currentUser.id).collection('followed')
+  .where('ref', '==', db.collection('users').doc(id))
+  .get()
+  .then((snapshot) => {
+    const { docs } = snapshot;
+    if (docs.length > 0) {
+      const item = docs[0];
+      return item.id;
+    }
+    return null;
+  });
+
+exports.onUnfollow = functions.firestore
+  .document('/users/{userId}/followers/{followerId}')
+  .onDelete(async (snapshot, context) => {
+    const { userId } = context.params;
+    const currentUser = snapshot.data().ref;
+    return db.collection('users').doc(currentUser.id).collection('followed').doc(await getFolloweddByUserId(userId, currentUser))
```

Comment:
```suggestion
    return currentUser.collection('followed').doc(getFolloweddByUserId(userId, currentUser))
```

Reply:
Dear Hugo,
the await is required
Regards,
Your favorite teammate

Reply:
not if youn remove you remove the get().then()